packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -389,7 +389,7 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
     }
 
     protected async createTerminal(): Promise<number> {
-        let rootURI = this.options.cwd;
+        let rootURI = this.options.cwd?.toString();
```

Comment:
@veredcon I'm not sure I understand the changes, `cwd` is already a `string | undefined`.

https://github.com/eclipse-theia/theia/blob/f8bcc19eba884cbd64112aeb691e8c97aad87197/packages/terminal/src/browser/base/terminal-widget.ts#L144-L147

The following changes gives a `string` if cwd is a `string` and `undefined` if it is not, so in reality this change does nothing?

Reply:
Hi @vince-fugnitto, 
Actually VSCode createTerminal API accept TerminalOptions as parameter, and there the cwd can be String | Uri:
https://code.visualstudio.com/api/references/vscode-api#TerminalOptions
![image](https://user-images.githubusercontent.com/9718939/109537716-e55ba080-7ac7-11eb-96a8-08821bf0f089.png)

The extension mentioned in the bug above, vscode-maven, is using the create terminal with URI:
`vscode.window.createTerminal({ name, env, cwd: <workspace folder URI passed>});`
Since they added this change:
https://github.com/microsoft/vscode-maven/pull/468/files it stopped working in theia (this is exactly what the bug mentions) 

So with this PR, you will support cases where extensions using the createTerminal API but the value passed is URI and not really a string.

Reply:
@veredcon I believe the change should then be done on the plugin side to also accept `Uri`.
At the moment, the plugin side only accepts `string | undefined` and the `toString` added as part of the changeset does not really make sense if we look at the typing.

https://github.com/eclipse-theia/theia/blob/f8bcc19eba884cbd64112aeb691e8c97aad87197/packages/plugin/src/theia.d.ts#L2781-L2784

Reply:
Thanks @vince-fugnitto , I fixed the typing, hope that's OK. Thanks!